detekt-test-utils/src/main/kotlin/io/github/detekt/test/utils/KotlinAnalysisApiEngine.kt

```diff
@@ -90,7 +95,7 @@ object KotlinAnalysisApiEngine {
         }
 
         try {
-            val file = session.modulesWithFiles.values.flatMap { it }.single() as KtFile
+            val file = session.modulesWithFiles.values.flatten().first { it.name == "dummy.kt" } as KtFile
```

Comment:
This should work and makes the intent slightly clearer IMHO:

```suggestion
            val file = session.modulesWithFiles.values.flatten().single { it.name == "dummy.kt" } as KtFile
```